pkg/gadgets/internal/networktracer/tracer.go

```diff
@@ -69,7 +74,25 @@ func newAttachment(
 		}
 	}()
 
-	a.collection, err = ebpf.NewCollection(spec)
+	spec = spec.Copy()
```

Comment:
A couple of eBPF questions 🙂 :

1) Why is this copy necessary?
2) Why we don't have to set the `current_netns` constant to the `SocketEnricher` specs, and it is ok to use the ones from dns and sni?

Reply:
> Why is this copy necessary?

The `spec` pointer comes from the caller of NewTracer() and we don't own it, so we should not modify it.

Also, if 2 containers are created at the same time, 2 calls to `newAttachment()` could be done in parallel and if the same variable `spec` was modified in a different way for the 2 attachments, they might not get the correct value for current_netns...


> Why we don't have to set the current_netns constant to the SocketEnricher specs, and it is ok to use the ones from dns and sni?

`#include <sockets-map.h>` is done both in the sni/dns modules and in the SocketEnricher module so they both have the variable `current_netns`. However that variable is only used in `gadget_socket_lookup()` which is a static inline function only used in sni/dns. So `current_netns` is not used in the SocketEnricher module. Maybe I should move the variable declaration in the `#ifndef SOCKETS_MAP_IMPLEMENTATION` to make that clear. In any case, `current_netns` is not meaningful in the SocketEnricher module since there is only one instance of the SocketEnricher module, whereas there are `n` instances of the sni/dns modules (`n` being the number of containers' netns).

Reply:
Thanks for the explanation. It is all clear now!

> Maybe I should move the variable declaration in the #ifndef SOCKETS_MAP_IMPLEMENTATION to make that clear.
>
I think it is fine like it is.